src/logger/entry.cr

```diff
@@ -0,0 +1,23 @@
+require "./severity"
+
+class Logger
+  class Entry
```

Comment:
`Entry` mutates itself so that message strings are evaluated at most once in the case of calls like `logger.debug{ do_expensive_thing.to_s }`. IMO this is a good semantic to preserve.

Reply:
I'm not sure that's worth it. Log messages are usually not very expensive and then it's much better to avoid allocations.

Reply:
I agree with @straight-shoota . It is tempting the idea of lazy building the logger message, but not at the cost of stressing the GC.

Reply:
I think lazy evaluation will save allocations in real-world scenarios, but putting them back in should be a non-breaking change, so I've removed them and will re-raise the issue in a later ticket if I can generate some meaningful benchmarks.

Reply:
Yeah, lazy evaluation itself really isn't the issue. It's memoizing the result...